deeplearning4j/deeplearning4j-core/src/test/java/org/deeplearning4j/nn/layers/samediff/testlayers/SameDiffConv.java

```diff
@@ -161,14 +160,51 @@ public void applyGlobalConfigToLayer(NeuralNetConfiguration.Builder globalConfig
 
     public static class Builder extends SameDiffLayer.Builder<Builder> {
 
+        @Getter
+        @Setter
         private int nIn;
+
+        @Getter
+        @Setter
         private int nOut;
+
+        @Getter
+        @Setter
         private Activation activation = Activation.TANH;
+
+        @Getter
         private int[] kernel = new int[]{2, 2};
+
+        public void setKernel(int[] kernel) {
+            Preconditions.checkArgument(kernel.length == 2, "Must have 2 kernel values - got %s", kernel);
```

Comment:
Also applies everywhere: for input validation - if we're adding it, let's have it in the one place for both. So, something like:
```
public void setKernel(int[] kernel) {
    kernel(kernel)
}
		
public Builder kernel(int... kernel){
    Preconditions.checkArgument(kernel.length == 2, "Must have 2 kernel values - got %s", kernel);
    this.kernel = kernel;
    return this;
}
```

Reply:
In some cases, the validation is only done in the constructor (e.g. ZeroPadding1DLayer).  I feel like moving the validation in those cases is a bit beyond the scope of this PR.

In the cases where there is a method to call, I'll use it.